docs/guides/expectations/how_to_create_and_edit_expectations_with_a_profiler.md

```diff
@@ -9,10 +9,230 @@ This guide will help you create a new Expectation Suite by profiling your data w
+    )
+```
+
+Alternatively, [you can instantiate a Data Context without a .yml file](../setup/configuring_data_contexts/how_to_instantiate_a_data_context_without_a_yml_file)
+
+
+### 2. Set your expectation_suite_name and create your Batch Request 
+
+The Batch Request specifies which batch of data you would like to profile in order to create your suite. We will pass it into a Validator in the next step.
+
+```python
+expectation_suite_name = "insert_the_name_of_your_suite_here"
+
+batch_request = {
+    "datasource_name": "my_datasource",
+    "data_connector_name": "default_inferred_data_connector_name",
+    "data_asset_name": "yellow_tripdata_sample_2020-05.csv",
+}
+```
+
+### 3. Instantiate your Validator
+
+We use a Validator to access and interact with your data. We will be passing the Validator to our Profiler in the next step.
+
+```python
+from great_expectations.core.batch import BatchRequest
+
+validator = context.get_validator(
+    batch_request=BatchRequest(**batch_request),
+    expectation_suite_name=expectation_suite_name
+)
+```
+
+After you get your Validator, you can call `validator.head()` to confirm that it contains the data that you expect.
+
+### 4. Instantiate a UserConfigurableProfiler
+
+Next, we instantiate a UserConfigurableProfiler, passing in the Validator with our data
+
+```python
+from great_expectations.profile.user_configurable_profiler import UserConfigurableProfiler
+profiler = UserConfigurableProfiler(profile_dataset=validator)
+```
+
+### 5. Use the profiler to build a suite
+
+Once we have our profiler set up with our batch, we call `profiler.build_suite()`. This will print a list of all the expectations created by column, and return the Expectation Suite object.
+
+```python
+suite = profiler.build_suite()
+```
+
+### 6. (Optional) Running validation, saving your suite, and building Data Docs
+
+If you'd like, you can validate your data with the new suite, save your Expectation Suite, and build Data Docs to take a closer look at the output
+
+```python
+from great_expectations.checkpoint.checkpoint import SimpleCheckpoint
+
+# Review and save our Expectation Suite 
+print(validator.get_expectation_suite(discard_failed_expectations=False))
+validator.save_expectation_suite(discard_failed_expectations=False)
+
+# Set up and run a Simple Checkpoint for ad hoc validation of our data 
+checkpoint_config = {
+    "class_name": "SimpleCheckpoint",
+    "validations": [
+        {
+            "batch_request": batch_request,
+            "expectation_suite_name": expectation_suite_name,
+        }
+    ],
+}
+checkpoint = SimpleCheckpoint(
+    f"_tmp_checkpoint_{expectation_suite_name}", context, **checkpoint_config
+)
+checkpoint_result = checkpoint.run()
+
+# Build and open Data Docs
+context.build_data_docs()
+
+validation_result_identifier = checkpoint_result.list_validation_result_identifiers()[0]
+context.open_data_docs(resource_identifier=validation_result_identifier)
```

Comment:
Is there a particular reason we're selecting index 0? Perhaps a comment to clarify?